tests/h/traversal/roots_test.py

```diff
@@ -19,6 +21,40 @@
 from h.traversal.contexts import AnnotationContext
 
 
+class TestRoot(object):
+    @pytest.mark.parametrize('permission', [
+        'admin_index',
+        'admin_groups',
+        'admin_mailer',
+        'admin_organizations',
+        'admin_users',
+    ])
+    def test_it_denies_all_permissions_for_unauthed_request(self, pyramid_config, pyramid_request, permission):
+        policy = pyramid.authorization.ACLAuthorizationPolicy()
+        pyramid_config.testing_securitypolicy(None)
+        pyramid_config.set_authorization_policy(policy)
+
+        context = Root(pyramid_request)
+
+        assert not pyramid_request.has_permission(permission, context)
+
+    @pytest.mark.parametrize('permission', [
+        'admin_index',
+        'admin_groups',
+        'admin_mailer',
+        'admin_organizations',
+        'admin_users',
+    ])
+    def test_it_allows_admin_permissions_for_requests_with_role(self, pyramid_config, pyramid_request, permission):
+        policy = pyramid.authorization.ACLAuthorizationPolicy()
+        pyramid_config.testing_securitypolicy('acct:adminuser@foo', groupids=[role.Admin])
+        pyramid_config.set_authorization_policy(policy)
+
+        context = Root(pyramid_request)
+
+        assert pyramid_request.has_permission(permission, context)
```

Comment:
I was a little confused by this second test. `groupids=[role.Admin]` is hard-coded, so what this is testing  is that **admins** are allowed all the permissions. Maybe the test should be called something like `test_it_allows_admin_permissions_for_admins`?

And I think there's a test missing for staff. If I delete all the `(Allow, role.Staff, "...")` lines from `Root` all the tests still pass.

I also noticed that if I delete the `DENY_ALL` from `Root` the tests still pass, but I think this may be because the `DENY_ALL` is in fact superfluous?